src/CodeGen_C.cpp

```diff
@@ -531,6 +531,29 @@ class ExternCallPrototypes : public IRGraphVisitor {
 };
 }
 
+void CodeGen_C::forward_declare_type(const Type &t) {
+    auto handle_type = t.handle_type;
+    if (!handle_type) return;
+    if (forward_declared.count(handle_type)) return;
+    auto type_type = handle_type->inner_name.cpp_type_type;
+    for (size_t ns = 0; ns < handle_type->namespaces.size(); ns++ ) {
+        stream << "namespace " << handle_type->namespaces[ns] << " {\n";
```

Comment:
In generated code, where one isn't coalescing namespace decls, I think it is ok to stack these on one line.